#! /usr/bin/env python

PACKAGE='ece6460_ulc_example'

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

steering_mode_list = gen.enum( [gen.const("YAW_RATE_MODE",   int_t, 0, ""),
                                gen.const("CURVATURE_MODE",  int_t, 1, "")], 
                                "Steering mode selection")

#       Name                Type       Lvl    Description                           Default    Min     Max
gen.add("enable_pedals",    bool_t,    0,     "Enable speed control of ULC",        True)
gen.add("enable_steering",  bool_t,    0,     "Enable steering control of ULC",     True)
gen.add("linear_velocity",  double_t,  0,     "Forward speed input",                0.0,       -4.5,  15.0)
gen.add("yaw_command",      double_t,  0,     "Curvature or yaw rate input",        0.0,       -2.0,  2.0)
gen.add("steering_mode",    int_t,     0,     "ULC steering mode",                  0, edit_method = steering_mode_list)

exit(gen.generate(PACKAGE, PACKAGE, "UlcExample"))
